Release v202211.0.0

MakisH opened the pull request:
Continuing with unreleased changes from https://github.com/precice/vm/pull/44.

## New changes

- Added ASTE and Julia bindings
- Added dune-fem
- Added the binaries of the elastic-tube-1d C++ solvers
- Upgraded OpenFOAM to v2206, CalculiX to v2.20, and Nutils to v7.2 (with a special case for the `perpendicular-flap/fluid-nutils` tutorial case)
- Added the Python pandas package
- Replaced the PDF reader Evince with Atril, fixing an issue with opening URLs
- Updated preCICE branch name (`master`->`main`)

## Previously unreleased changes

- Added libvirt support (experimental)
- Updated a few CI-related things

fsimonis commented:
aste-turbine: :x:
`precice-aste-evaluate: command not found`

channel-transport/fluid-nutils/run.sh: :x:

```
opened log at file:///home/vagrant/public_html/fluid.py/log-1.html
nutils v6.3 "garak-guksu"
start Mon Nov 21 11:22:59 2022
Running utils
ValueError: not a 'str': ['u']
```

channel-transport-reaction: :o: 

Super skewed though. fluid reaches t= 5/5 while the chemical part is still at t=1.25/5

elastic-tube-1d: python :o: (cpp untested)

elastic-tube-3d:
fluid-openfoam + solid-calculix ...
fluid-openfoam + solid-fenics  ...

MakisH commented:
@fsimonis thanks for testing! I fixed several issues (including the aste and nutils) and restarted the build.

> Super skewed though. fluid reaches t= 5/5 while the chemical part is still at t=1.25/5

This is fine, right? It is a uni-directional coupling.

> elastic-tube-3d:
> fluid-openfoam + solid-calculix :o: (plot displacements doesn't work though)

What is the error you get? Did you run both cases before running the plotting script? In any case, this would be something for the tutorials. 

> fluid-openfoam + solid-fenics :o:

The `solid-fenics` case does not seem to work for me, at least in my now modified VM (which should not have anything relevant):

```bash
vagrant@precicevm:~/tutorials/elastic-tube-3d/solid-fenics$ ./run.sh 
Traceback (most recent call last):
  File "solid.py", line 2, in <module>
    from fenics import Constant, Function, AutoSubDomain, VectorFunctionSpace, interpolate, \
  File "/usr/lib/python3/dist-packages/fenics/__init__.py", line 7, in <module>
    from dolfin import *
  File "/usr/lib/petsc/lib/python3/dist-packages/dolfin/__init__.py", line 144, in <module>
    from .fem.assembling import (assemble, assemble_system, assemble_multimesh, assemble_mixed,
  File "/usr/lib/petsc/lib/python3/dist-packages/dolfin/fem/assembling.py", line 38, in <module>
    from ufl.form import sub_forms_by_domain
ImportError: cannot import name 'sub_forms_by_domain' from 'ufl.form' (/home/vagrant/.local/lib/python3.8/site-packages/ufl/form.py)
```

This seems to be working in v2202, so, if you can reproduce it, it will be a regression.

MakisH commented:
Regarding the plotting script, this does not seem to be a regression: https://github.com/precice/vm/issues/63

MakisH commented:
With the latest build:
- All FEniCS cases fail to start :x:, with the same error as in my previous comment. @IshaanDesai any clue why I may be getting this message? This is definitely a new VM issue now. Have a look at the recent commits as well.
 
Removing `pandas` did not help. Removing the `aste` directory from PATH did not help (maybe there was some metis conflict). Removing `dune-fem` did not help. Removing `atril` did not help. Removing `nutils` did not help. Removing `python3.8-venv` did not help.

Is maybe just FEniCS broken right now and I got a bad update?

```bash
vagrant@precicevm:~/tutorials/flow-over-heated-plate/solid-fenics$ ./run.sh 
No protocol specified
Traceback (most recent call last):
  File "solid.py", line 6, in <module>
    from fenics import Function, SubDomain, RectangleMesh, BoxMesh, FunctionSpace, VectorFunctionSpace, Point, \
  File "/usr/lib/python3/dist-packages/fenics/__init__.py", line 7, in <module>
    from dolfin import *
  File "/usr/lib/petsc/lib/python3/dist-packages/dolfin/__init__.py", line 144, in <module>
    from .fem.assembling import (assemble, assemble_system, assemble_multimesh, assemble_mixed,
  File "/usr/lib/petsc/lib/python3/dist-packages/dolfin/fem/assembling.py", line 38, in <module>
    from ufl.form import sub_forms_by_domain
ImportError: cannot import name 'sub_forms_by_domain' from 'ufl.form' (/home/vagrant/.local/lib/python3.8/site-packages/ufl/form.py)
```

- `aste-turbine` completes
- `channel-transport` Nutils cases start
- `flow-over-heated-plate/solid-dunefem` starts
- `flow-over-heated-plate/solid-nutils` starts
- `perpendicular-flap/fluid-nutils` starts (with Nutils 6.3 automatically set)
- the `elastic-tube-1d` C++ cases start
- the plotting script of the `oscillator` works
- URLs from PDF files open normally in Firefox

Other than FEniCS, I am triggering a new build to get the new tutorials, and this is ready.

davidscn commented on provisioning/install-aste.sh:
```diff
@@ -0,0 +1,23 @@
+)
+
+# Add aste to PATH and libmetis to the library path
+echo "export PATH=\"\${HOME}/aste/build:\${PATH}\"" >>~/.bashrc
+echo "export LD_LIBRARY_PATH=\"\${HOME}/aste/build:\${LD_LIBRARY_PATH}\"" >>~/.bashrc
```
In principle, we could also use `make install` to install these executables system-wide.

MakisH replied:
True. The reason I avoid that (and that I use repositories) is to let people easily `git pull` newer versions, if needed. I find that `make install` on a non-standard prefix (especially not via a package manager) makes this confusing, as sometimes users think that they are using one version, while they are using another.

IshaanDesai commented:
The error seen with FEniCS is a known issue and has been mentioned in the [troubleshooting section](https://github.com/precice/fenics-adapter#troubleshooting) of the fenics-adapter. I am trying to see if the solution written there also works for the vm, will open a PR if that is the case.

IshaanDesai reviewed (approved):
I ran the FEniCS cases in a VM created from the artifact of https://github.com/precice/vm/actions/runs/3524369114 and the cases work.

MakisH commented:
I confirm that the FEniCS cases now also run. Thank you very much! :hugs:
